.gitlab/deps_build.yml

```diff
@@ -9,8 +9,11 @@
   script:
     # use tmpdir to prevent git remote capture by clang build
     - mkdir /tmp/clangbuild && cd /tmp/clangbuild
-    - wget https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.1/clang-11.0.1.src.tar.xz -O clang.src.tar.xz
-    - wget https://github.com/llvm/llvm-project/releases/download/llvmorg-11.0.1/llvm-11.0.1.src.tar.xz -O llvm.src.tar.xz
+    - CLANG_SHA256SUM=73f572c2eefc5a155e01bcd84815751d722a4d3925f53c144acfb93eeb274b4d
+    - LLVM_SHA256SUM=ccd87c254b6aebc5077e4e6977d08d4be888e7eb672c6630a26a15d58b59b528
+    - wget ${GENERAL_ARTIFACTS_CACHE_BUCKET_URL}/llvm/clang-11.0.1.src.tar.xz -O clang.src.tar.xz
```

Comment:
I think caching this is a bit redundant. Note this is a manual job that is builds LLVM and then caches the resulting (pre-built) artifact in S3. We only use it once when we want to update LLVM, and that means we would have to change the URL anyway. During the actual Agent build, the LLVM artifact that is used already comes from an S3 bucket.
+1 to having checksums checks, but caching the sources here seems redundant to me. Also, it adds one additional step every time we update LLVM, since we would have to first cache the sources to run this only once.

Reply:
I see your point and to a certain extent I agree; here are my counterpoints:

* We upgrade very rarely (last time Feb 5 2021, so more than a year ago).
* We would also need to rerun this if we wanted to change the build job if we e.g. needed to change the cmake args.

I think it's still useful to have *everything* cached and I think given the rareness of updates of clang/llvm it's ok to keep the change as-is.